src/interpreter.rs

```diff
@@ -192,45 +184,48 @@ impl<T: Clone> Stack<T> {
         Stack(vec![])
     }
 
-    fn reverse_index(&self, i: isize) -> Result<usize, ScriptError> {
-        usize::try_from(-i)
-            .map(|a| self.0.len() - a)
-            .map_err(|_| ScriptError::InvalidStackOperation)
+    fn rindex(&self, i: usize) -> Result<usize, ScriptError> {
+        let len = self.0.len();
+        if i < len {
+            Ok(len - i - 1)
+        } else {
+            Err(ScriptError::InvalidStackOperation)
+        }
     }
 
-    pub fn top(&self, i: isize) -> Result<&T, ScriptError> {
-        let idx = self.reverse_index(i)?;
+    pub fn rget(&self, i: usize) -> Result<&T, ScriptError> {
+        let idx = self.rindex(i)?;
         self.0.get(idx).ok_or(ScriptError::InvalidStackOperation)
     }
 
-    pub fn swap(&mut self, a: isize, b: isize) -> Result<(), ScriptError> {
-        let au = self.reverse_index(a)?;
-        let bu = self.reverse_index(b)?;
-        self.0.swap(au, bu);
+    pub fn rswap(&mut self, a: usize, b: usize) -> Result<(), ScriptError> {
```

Comment:
`Stack` is public, so all of these renames are breaking changes. Document them as such in the changelog.